rustual-boy-cli/src/argparse.rs

```diff
@@ -0,0 +1,32 @@
+use clap::{App, Arg};
+
+pub struct CommandLineConfig {
+    pub rom_path: String,
+    pub sram_path: String,
+}
+
+pub fn parse_args() -> CommandLineConfig {
+    let app = App::new("Rustual-boy")
+        .version("0.1.0")
+        .author(crate_authors!(", "))
+        .about("A CLI frontend to the rustual-boy emulator")
+        .arg(Arg::with_name("ROM")
+             .help("The name of the ROM to load")
+             .required(true)
+             .index(1)
+        ).arg(Arg::with_name("SRAM")
+              .help("Path to an SRAM")
+              .short("s")
+              .long("sram")
+        );
+
+    let matches = app.get_matches();
+
+    CommandLineConfig {
+        rom_path: matches.value_of("ROM").unwrap().into(),
```

Comment:
That `unwrap` is actually slightly misleading, `clap` will actually never return a `None` here because "ROM" is marked as required. Instead it prints the help text and exits. `clap` will do the same thing for any argument validation errors. I'll leave a comment to that effect.